tests/algorithm/scripts/expand_test_data.py

```diff
@@ -0,0 +1,243 @@
+                        # generating the appropriate indices
+                        for pos in range(len(safe_spots) - 1):
+                            if safe_spots[pos] == safe_spots[pos + 1] - 1:
+                                eligible_swap = True
+                                break
+                        
+                        # There's at least one place where we can switch two adjacent
+                        # characters, so randomly pick one such spot
+                        if eligible_swap:
+                            valid_trans = False
+                            left_swap_at = 0
+                            right_swap_at = 0
+                            while not valid_trans:
+                                left_swap_at = random.choice(safe_spots)
+                                right_swap_at = left_swap_at + 1
+                                if right_swap_at in safe_spots:
+                                    valid_trans = True
+
+                            altered_positions.append(left_swap_at)
+                            altered_positions.append(right_swap_at)
+                            edits_to_apply.append(("transpose", left_swap_at))
+                            made_edit = True
+
+        # Now apply edits to this field
+        # We'll work in reverse order of index to avoid interference
+        edits_to_apply = sorted(edits_to_apply, key=lambda x: x[1], reverse=True)
+        for e in edits_to_apply:
+            if e[0] == "add":
+                if e[1] < len(base_value):
+                    base_value = base_value[:e[1]] + e[2] + base_value[e[1]:]
+                # Add to the end, just concatenate
+                else:
+                    base_value += e[2]
+            
+            # Deleting is just splicing out the character
+            elif e[0] == "delete":
+                if e[1] == len(base_value) - 1:
+                    base_value = base_value[:len(base_value) - 1]
+                else:
+                    base_value = base_value[:e[1]] + base_value[e[1] + 1:]
+
+            # Python strings are immutable and can't be assigned, so just
+            # reconstruct a new string with the letters manually inserted
+            else:
+                left_char = base_value[e[1]]
+                right_char = base_value[e[1] + 1]
+                base_value = base_value[:e[1]] + right_char + left_char + base_value[e[1] + 2:]
+        test_case[f] = base_value
+    
+    return test_case
+
+
+# Start by reading in the file
+test_cases = []
+with open(BASE_TEST_FILE, 'r') as fp:
+    test_cases = fp.readlines()
+
+# Now, let's do a bit of formatting and parsing:
+# Each case will be a dict mapping a formatted value to the
+# appropriate header
+headers = [h.strip() for h in test_cases[0].split(",")]
+test_cases = test_cases[1:]
+# Some field values are double-quoted strings which themselves contain commas
```

Comment:
Just so you know, the [csv module has a reader](https://docs.python.org/3/library/csv.html#csv.reader) that does this same logic.